### ISSUE TITLE
Deleting a Rooted Object

### ISSUE THREAD

**Issue by jmirtsch** [jmirtsch/GitIFCPrototype#4]
First implementation simply assigns an owner history with a deleted state enumeration set.
This isn't particularly intuitive or obvious in a human review of the model and needs to be considered.

**Comment by mtveerman** [comment]
This is interesting. We could have a discussion about the fact that, when we have a solid versioning system in place for ifc, things like owner history state are no longer necessary. The versioning system takes care of that.

**Comment by radumg** [comment]
True, but as far as I understand (and do correct me if i'm wrong) the owner history state could more data-rich that the git commit history, no ?

**Comment by jmirtsch** [comment]
Yes, the owner history is more data rich, so I'm preserving it for the time being.  One option is to include a flag in the element folder name (ie "0GDokRCPb3nPU27Rv5we1H Beam DELETED")  An alternative is to move the object into a nominated folder.
These could be purged from time to time (ie project milestones or after issues).  
I like the idea of retaining the object description so you can have a visualization of model history over time without trawling back through git commits.  Then you have the best of both worlds with git history and model specific history.

**Comment by mtveerman** [comment]
Marking folders with a DELETED tag is not a good idea (where would that end?). 

I believe you think too much of the repository being human readable. If you step away from that, and consider the repository to be only used by a processor, than there is nothing wrong with simply keeping the object as is, with the correct link to the ownerhistory which in turn shows how the object was modified.
